src/Context/AuthenticationProvider.jsx

```diff
@@ -0,0 +1,80 @@
+import { createContext, useContext, useReducer, useState } from "react";
+import axios from "axios";
+import { userCredReducer } from "../Reducer/auth-reducer";
+const AuthContext = createContext();
+
+export const useAuthContext = () => useContext(AuthContext);
+
+export const initialUserState = { name:"", username:"", password:"", email:"" };
+
+export const AuthenticationProvider = ({ children }) => {
+  const [login, setLogin] = useState(localStorage.getItem("login") || false);
+  const [userState, userDispatch] = useReducer(userCredReducer, initialUserState);
+  const [userData, setUser] = useState(JSON.parse(localStorage.getItem("userData")) || {});
+  const [showLoader, setShowLoader] = useState(false);
+
+  const loginUser = async (name, pwd) => {
+    try {
+      const { data } = await axios.post("https://api-supminn.herokuapp.com/users/login", {
```

Comment:
You can define the address "https://api..." as a constant & use it in the axios calls.